arelle/plugin/validate/ROS/rules/ros.py

```diff
@@ -293,6 +292,29 @@ def rule_main(
     modelXbrl.modelManager.showStatus(None)
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+)
+def rule_ros19(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    ROS: Rule 19: DPLTurnoverRevenue should be tested to be less than or equal to 10x the absolute value of Equity
+    """
+    equity_fact = next(iter(val.modelXbrl.factsByLocalName.get(EQUITY, set())))
+    turnover_fact = next(iter(val.modelXbrl.factsByLocalName.get(TURNOVER_REVENUE, set())))
```

Comment:
These will throw exceptions if there are no facts for these concepts.

If there is more than one, should we validate all of them?

Reply:
Im not 100% sure. Im trying to get more info from Peter.